src/Hl7.Fhir.Base/Serialization/BaseFhirJsonPocoDeserializer.cs

```diff
@@ -335,6 +342,142 @@ private void deserializeObjectInto<T>(
             PocoDeserializationHelper.RunInstanceValidation(target, Settings.Validator, context, state.Errors);
         }
     }
+    
+    private void deserializeUnknownPropertiesInto<T>(
+        T target,
+        string propertyName,
+        ref Utf8JsonReader reader,
+        FhirJsonPocoDeserializerState state,
+        bool stayOnLastToken = false) where T : Base
+    {
+        target.TryGetValue(propertyName, out var value);
+
+        var (name, choiceType) = tryDetectChoiceTypeFromName(propertyName);
+
+        // move past property name
+        reader.Read();
+
+        object? result;
+        if(choiceType is not null)
+        {
+            state.Path.EnterElement(name, null, choiceType.IsPrimitive);
+            
+            result = DeserializeFhirPrimitive(value as PrimitiveType, name, choiceType, null, ref reader, new(), state);
```

Comment:
This does not need to be a primitive, right? Choice types can be non-primitives too,